calculation/gmhazard_calc/gmhazard_calc/gms/GroundMotionDataset.py

```diff
@@ -241,34 +246,52 @@ def get_im_df(
         sf: pd.Series = None,
     ) -> pd.DataFrame:
         """See GMDataset method for parameter specifications"""
-        im_df = self._im_df.copy()
+        im_df = self._im_df.copy().loc[:, IMs]
+        metadata_df = self.get_metadata_df(site_info)
+
+        # CS Param bounds filtering
+        if cs_param_bounds is not None:
+            mask = (
+                np.ones(im_df.shape[0], dtype=bool)
+                if cs_param_bounds is None
+                else self._get_filter_mask(
+                    metadata_df.loc[im_df.index], cs_param_bounds
+                )
+            )
+            im_df = im_df.loc[mask]
```

Comment:
You've got an extra check here for cs_param_bounds being None so it will always do the get_filter_mask in this if statement